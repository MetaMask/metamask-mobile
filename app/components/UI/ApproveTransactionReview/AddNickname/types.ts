export interface AddNicknameProps {
  closeModal: () => void;
  address: string;
  addressNickname: string;
  nicknameExists: boolean;
  showModalAlert: (config: any) => void;
  networkState: {
    provider: {
      type: string;
    };
    network: number;
  };
}
