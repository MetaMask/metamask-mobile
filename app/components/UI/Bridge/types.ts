import { TxData, Quote } from '@metamask/bridge-controller';
import type { CaipChainId } from '@metamask/utils';

export interface BridgeToken {
  address: string;
  symbol: string;
  image: string;
  decimals: number;
  chainId: CaipChainId;
}

// TODO: use type from @metamask/bridge-controller once "approval" is made optional
export interface QuoteResponse {
  quote: Quote;
  approval?: TxData | null;
  trade: TxData;
  estimatedProcessingTimeInSeconds: number;
}
