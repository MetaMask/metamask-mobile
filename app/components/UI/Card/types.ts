import { ethers } from 'ethers';
import { FlashListAssetKey } from '../Tokens/TokenList';

/**
 * Enum for asset delegation status
 */
export enum AllowanceState {
  NotActivated = 'not_activated',
  Unlimited = 'unlimited',
  Limited = 'limited',
}

// Helper interface for token balances
export interface CardToken {
  address: string | null;
  decimals: number | null;
  symbol: string | null;
  name: string | null;
}

export type CardTokenAllowance = {
  allowanceState: AllowanceState;
  allowance: ethers.BigNumber;
} & FlashListAssetKey &
  CardToken;
