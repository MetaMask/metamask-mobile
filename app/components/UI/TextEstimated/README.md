# TextEstimated

TextEstimated is a [Text](../Text/Text.tsx) component with a prefix Tilde [Icon](../../Icon/Icon.tsx).

## Props

This component extends [TextProps](../Text/Text.types.ts#L33).

### `variant`

Optional enum to select between Typography variants.

| <span style="color:gray;font-size:14px">TYPE</span> | <span style="color:gray;font-size:14px">REQUIRED</span> | <span style="color:gray;font-size:14px">DEFAULT</span> |
| :-------------------------------------------------- | :------------------------------------------------------ | :----------------------------------------------------- |
| [TextVariant](./Text.types.ts#L6)                   | No                                                      | BodyMD                                                |

## Usage

```javascript
// Replace import with relative path.
import TextEstimated from 'app/components/UI/TextEstimated';
import { TextVariant } from 'app/component-library/components/Texts/Text';

<TextEstimated variant={TextVariant.HeadingSMRegular}>
    {SAMPLE_TEXT}
</TextEstimated>;
```
