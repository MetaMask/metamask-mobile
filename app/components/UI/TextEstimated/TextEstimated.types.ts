// External dependencies.
import { IconSize } from '../../../component-library/components/Icon';
import {
  TextProps,
  TextVariant,
} from '../../../component-library/components/Texts/Text/Text.types';

/**
 * TextEstimated component props.
 */
export type TextEstimatedProps = TextProps;

/**
 * Mapping of IconSize by TextVariant.
 */
export type TildeIconSizeByTextVariant = {
  [key in TextVariant]: IconSize;
};
