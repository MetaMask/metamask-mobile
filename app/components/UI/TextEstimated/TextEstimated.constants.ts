/* eslint-disable import/prefer-default-export */
// Internal dependencies.
import { TildeIconSizeByTextVariant } from './TextEstimated.types';

// External dependencies.
import { IconSize } from '../../../component-library/components/Icon';
import { TextVariant } from '../../../component-library/components/Texts/Text/Text.types';

export const TILDE_ICON_SIZE_BY_TEXT_VARIANT: TildeIconSizeByTextVariant = {
  [TextVariant.DisplayMD]: IconSize.Lg,
  [TextVariant.HeadingLG]: IconSize.Md,
  [TextVariant.HeadingMD]: IconSize.Sm,
  [TextVariant.HeadingSMRegular]: IconSize.Xs,
  [TextVariant.HeadingSM]: IconSize.Sm,
  [TextVariant.BodyMD]: IconSize.Xs,
  [TextVariant.BodyMDBold]: IconSize.Xs,
  [TextVariant.BodySM]: IconSize.Xss,
  [TextVariant.BodySMBold]: IconSize.Xss,
  [TextVariant.BodyXS]: IconSize.Xss,
};

export const TEST_SAMPLE_TEXT = 'Sample Text';

export const TEXT_ESTIMATED_TEST_ID = 'text-estimated';
