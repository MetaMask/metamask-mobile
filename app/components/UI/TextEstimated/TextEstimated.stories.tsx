// Third party dependencies.
import React from 'react';
import { text, select } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';

// External dependencies.
import { TextVariant } from '../../../component-library/components/Texts/Text/Text.types';

// Internal dependencies.
import TextEstimated from './TextEstimated';
import { TEST_SAMPLE_TEXT } from './TextEstimated.constants';

storiesOf('Components / UI / TextEstimated', module).add('Default', () => {
  const groupId = 'Props';
  const textVariantSelector = select(
    'Variant',
    TextVariant,
    TextVariant.HeadingSMRegular,
    groupId,
  );
  const inputText = text('Text', TEST_SAMPLE_TEXT, groupId);

  return (
    <TextEstimated variant={textVariantSelector}>{inputText}</TextEstimated>
  );
});
