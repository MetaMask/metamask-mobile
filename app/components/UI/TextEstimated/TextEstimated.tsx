/* eslint-disable react/prop-types */

// Third party dependencies.
import React from 'react';

// External dependencies.
import {
  IconName,
  IconProps,
} from '../../../component-library/components/Icon';
import TextWithPrefixIcon from '../../../component-library/components/Texts/TextWithPrefixIcon';
import { TextVariant } from '../../../component-library/components/Texts/Text/Text.types';

// Internal dependencies.
import {
  TILDE_ICON_SIZE_BY_TEXT_VARIANT,
  TEXT_ESTIMATED_TEST_ID,
} from './TextEstimated.constants';
import { TextEstimatedProps } from './TextEstimated.types';

const TextEstimated: React.FC<TextEstimatedProps> = ({
  variant = TextVariant.BodyMD,
  children,
  ...props
}) => {
  const iconProps: IconProps = {
    name: IconName.TildeOutline,
    size: TILDE_ICON_SIZE_BY_TEXT_VARIANT[variant],
  };
  return (
    <TextWithPrefixIcon
      iconProps={iconProps}
      variant={variant}
      testID={TEXT_ESTIMATED_TEST_ID}
      {...props}
    >
      {children}
    </TextWithPrefixIcon>
  );
};

export default TextEstimated;
