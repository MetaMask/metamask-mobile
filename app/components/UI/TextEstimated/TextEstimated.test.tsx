// Third party dependencies.
import React from 'react';
import { shallow } from 'enzyme';

// External dependencies.
import { TextVariant } from '../../../component-library/components/Texts/Text/Text.types';

// Internal dependencies.
import TextEstimated from './TextEstimated';
import {
  TEXT_ESTIMATED_TEST_ID,
  TEST_SAMPLE_TEXT,
} from './TextEstimated.constants';

describe('TextEstimated - Snapshot', () => {
  it('should render default settings correctly', () => {
    const wrapper = shallow(
      <TextEstimated variant={TextVariant.HeadingSMRegular}>
        {TEST_SAMPLE_TEXT}
      </TextEstimated>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('TextEstimated', () => {
  it('should render TextEstimated', () => {
    const wrapper = shallow(
      <TextEstimated variant={TextVariant.HeadingSMRegular}>
        {TEST_SAMPLE_TEXT}
      </TextEstimated>,
    );
    const TextEstimatedComponent = wrapper.findWhere(
      (node) => node.prop('testID') === TEXT_ESTIMATED_TEST_ID,
    );
    expect(TextEstimatedComponent.exists()).toBe(true);
  });
});
