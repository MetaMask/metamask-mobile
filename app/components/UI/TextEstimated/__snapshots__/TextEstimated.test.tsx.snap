// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextEstimated - Snapshot should render default settings correctly 1`] = `
<TextWithPrefixIcon
  iconProps={
    Object {
      "name": "TildeOutline",
      "size": "12",
    }
  }
  testID="text-estimated"
  variant="sHeadingSMRegular"
>
  Sample Text
</TextWithPrefixIcon>
`;
